src/python/pants/pantsd/pants_daemon.py

```diff
@@ -39,7 +39,13 @@
 
 
 class _LoggerStream(object):
-  """A sys.{stdout,stderr} replacement that pipes output to a logger."""
+  """A sys.std{out,err} replacement that pipes output to a logger.
+
+  N.B. the Logger object expects unicode.
```

Comment:
I can't figure out how to insert a multi-line suggestion, but I wouldn't leave this line alone from the rest, and would fill it to just:
```
  N.B. the Logger object expects unicode. However, most of our outstream logic, such as in
  `Exiter.py`, will use `sys.std{out,err}.buffer` and thus a bytes interface when running with
  Python 3. So, we must provide a `buffer` property, and change the semantics of the buffer to
  always convert the message to unicode.
```